backend/stack_overflow/src/main/java/com/gabrielcireap/stackOverflow/service/QuestionManagementService.java

```diff
@@ -27,115 +24,91 @@
-
-        User user = new User();
-        user.setId(questionDTO.getUser().getId());
-        user.setUsername(questionDTO.getUser().getUsername());
-        user.setScore(questionDTO.getUser().getScore());
-        user.setIsAdmin(questionDTO.getUser().getIsAdmin());
-        user.setIsBanned(questionDTO.getUser().getIsBanned());
-        question.setUser(user);
-
-        question.setId(repositoryFactory.createQuestionRepository().save(question).getId());
-        QuestionDTO dto =  QuestionDTO.ofEntity(question);
-        System.out.println(dto);
-        return dto;
+        QuestionDTO savedQuestion = QuestionDTO.ofEntity(repositoryFactory.createQuestionRepository().save(question));
+        eventPublisher.publishEvent(new QuestionUpdatedEvent(savedQuestion));
+        return savedQuestion;
     }
 
     @Transactional
     public void remove(int questionId) {
-
-        userManagementService.checkIfUserIsLogged();
         if (!userManagementService.getLoggedUser().getIsAdmin()) {
             throw new NotEnoughPermissionsException();
         }
-
         Question question = repositoryFactory.createQuestionRepository().findById(questionId).orElseThrow(QuestionNotFoundException::new);
         repositoryFactory.createQuestionRepository().remove(question);
+        eventPublisher.publishEvent(new QuestionDeletedEvent(QuestionDTO.ofEntity(question)));
     }
 
-    /*@Transactional
-    public void edit(int questionId, String title, String text){
-        userManagementService.checkIfUserIsLogged();
-        if(!userManagementService.getLoggedUser().getIsAdmin()){
+    @Transactional
+    public QuestionDTO edit(int questionId, String title, String text) {
+        if (!userManagementService.getLoggedUser().getIsAdmin()) {
             throw new NotEnoughPermissionsException();
         }
-
-        Question question = (Question) findById(questionId).keySet().toArray()[0];
+        Question question = repositoryFactory.createQuestionRepository().findById(questionId).orElseThrow(QuestionNotFoundException::new);
         question.setTitle(title);
         question.setText(text);
-        save(question);
-    }*/
+        QuestionDTO questionDTO = save(QuestionDTO.ofEntity(question));
+        eventPublisher.publishEvent(new QuestionUpdatedEvent(questionDTO));
+        return questionDTO;
+    }
 
     @Transactional
     public List<QuestionDTO> findQuestionByTitle(String text) {
-
-        //userManagementService.checkIfUserIsLogged();
         List<Question> questions = repositoryFactory.createQuestionRepository().findByTitle(text);
-        questions.sort((q1, q2) -> {
-            return q1.getCreationDate().after(q2.getCreationDate()) ? -1 : 1;
-        });
-        System.out.println(questions);
-        return questions.stream().map(QuestionDTO::ofEntity).collect(Collectors.toList());
+        questions.sort((q1, q2) -> q1.getCreationDate().after(q2.getCreationDate()) ? -1 : 1);
```

Comment:
If a class attribute (e.g. `creationDate`) implements the `Comparable` interface, you can avoid creating a lambda by using `Comparator.comparing(Question::getCreationDate)`